actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -45,7 +45,11 @@ def action
       end
 
       def internal?
-        controller.to_s =~ %r{\Arails/(info|mailers|welcome)} || path =~ %r{\A#{Rails.application.config.assets.prefix}\z}
+        internal_controller = controller.to_s =~ %r{\arails/(info|mailers|welcome)}
+        internal_asset = Rails.application.config.respond_to?(:assets) &&
+          path =~ %r{\a#{Rails.application.config.assets.prefix}\z}
+
+        internal_controller || internal_asset
```

Comment:
The original is using the `||` short-circuit in a much nicer way then here.
You do all the work for `internal_asset` even though you don't need it.

A nicer solution might be to split it into two private methods and call them with the same short-circuiting logic:

```ruby
def internal?
  internal_controller? || internal_asset?
end

...

private

def internal_controller
  controller.to_s =~ %r{\arails/(info|mailers|welcome)}
end

def internal_asset
  Rails.application.config.respond_to?(:assets) &&
    path =~ %r{\a#{Rails.application.config.assets.prefix}\z}
end
```